kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ServiceOperationsImpl.java

```diff
@@ -83,4 +79,37 @@ public Service waitUntilReady(long amount, TimeUnit timeUnit) throws Interrupted
 
     return get();
   }
+
+  public String getURL(String portName) {
+    String clusterIP =  getMandatory().getSpec().getClusterIP();
+    if("None".equals(clusterIP)) {
+      throw new IllegalStateException("Service: " + getMandatory().getMetadata().getName() + " in namespace " +
+        namespace + " is head-less. Search for endpoints instead");
+    }
+
+    ServiceLoader<ServiceToURLProvider> urlProvider = ServiceLoader.load(ServiceToURLProvider.class, Thread.currentThread().getContextClassLoader());
+    Iterator<ServiceToURLProvider> iterator = urlProvider.iterator();
+    List<ServiceToURLProvider> servicesList = new ArrayList<>();
+
+    while(iterator.hasNext()) {
+      servicesList.add(iterator.next());
+    }
+
+    // Sort all loaded implementations according to priority
+    Collections.sort(servicesList, new ServiceToUrlSortComparator());
```

Comment:
I will add from line 90 to 99 into a method that gets and order services, so the method `getURL` looks more compact.